File: plus_one.cpp
vector<int> plusOne(vector<int>& digits) {
    if(digits.empty()){
        vector<int> v(1,1);
        return v;
    }
    if(digits.back()+1 == 10){
        digits.pop_back();
        digits = plusOne(digits);
        digits.push_back(0);
    }
    else{
        digits.back()++;
    }
    return digits;
}

File: plus_one.py
def plusOne(self, digits: List[int]) -> List[int]:
    if digits==[]:
        return [1]
    if digits[-1]+1 == 10:
        digits = self.plusOne(digits[:len(digits)-1])
        digits.append(0)
    else:
        digits[-1]+=1
    return digits
        
